### ISSUE TITLE
Validator::allowedValues() doesn't respect Validator::ifPresent()

### ISSUE THREAD

**Issue by ryan-lange** [vlucas/phpdotenv#418]
When using ifPresent() followed by allowedValues(), a ValidationException will always be thrown if the environment variable is missing.

The following example assumes no `.env` file.

### Example
```
<?php

use Dotenv\Dotenv;

require __DIR__ . '/vendor/autoload.php';

$dotenv = Dotenv::createMutable( __DIR__ );
$dotenv->safeLoad();

$dotenv->ifPresent( 'FOO' )
       ->allowedValues( [ 'bar', 'baz' ] );
```

### Actual Output
```
PHP Fatal error:  Uncaught Dotenv\Exception\ValidationException: One or more environment variables failed assertions: FOO is not one of [bar, baz]. in [...]/phpdotenv/vendor/vlucas/phpdotenv/src/Validator.php:181
```

### Expected Output
Nothing.

### Version
4.1.2

**Comment by GrahamCampbell** [comment]
Thanks for getting in touch, but we already have a test that confirms this is not the case, and no exception will be thrown: https://github.com/vlucas/phpdotenv/blob/master/tests/Dotenv/ValidatorTest.php#L36-L42.

Note that there is a distinction between being not present, and being the empty string. The empty string is considered to be present, and set to the empty string, which is different from being unset.

**Comment by GrahamCampbell** [comment]
Interesting thanks. Related failing test: https://github.com/vlucas/phpdotenv/pull/419.

**Comment by GrahamCampbell** [comment]
This will be fixed in the upcoming `v3.6.2` and `v4.1.3` releases, which should be released within the next few hours.
